Checkers/Board.cs

```diff
@@ -0,0 +1,272 @@
+        }
+
+
+        // main method invoked for input and moving checkers no board
+        public void moveCheckerOnBoard()
+        {
+
+            int oldRow = 0;
+            int oldColumn = 0;
+            int newRow = 0;
+            int newColumn = 0;
+
+            Console.WriteLine("Enter pick up row.");
+            oldRow = int.Parse(Console.ReadLine());
+            Console.WriteLine("Enter pick up column.");
+            oldColumn = int.Parse(Console.ReadLine());
+
+            Console.WriteLine("Enter placement row.");
+            newRow = int.Parse(Console.ReadLine());
+            Console.WriteLine("Enter placement column.");
+            newColumn = int.Parse(Console.ReadLine());
+
+            moveChecker(oldRow, oldColumn, newRow, newColumn);
+            reloadBoard();
+            
+
+        }
+
+        // method called to move checker from prior position
+        public void moveChecker(int oldR, int oldC, int newR, int newC)
+        {
+            int oldRow = oldR;
+            int oldColumn = oldC;
+            int newRow = newR;
+            int newColumn = newC;
+            Checker checkerMoved ;
+            Checker placedChecker ;
+
+
+            checkerMoved = SelectChecker(oldRow, oldColumn);
+            placedChecker = SelectChecker(newRow,newColumn);
+
+            //checks to make sure it is a valid checker and a valid move
+            if(Checkers.Contains(checkerMoved) && !Checkers.Contains(placedChecker))
+            {
+                int [] newPosition = new int[]{ newRow, newColumn};
+                checkerMoved.Position = newPosition;
+                RemoveChecker(oldRow, oldColumn);
+            }
+
+            
+
+        }
+
+        // method invoked to get input and remove checker from the board
+        public void removeCheckerFromBoard()
+        {
+            int oldRow = 0;
+            int oldColumn = 0;
+            Checker validationChecker;
+
+            Console.WriteLine("Enter row to remove.");
+            oldRow = int.Parse(Console.ReadLine());
+            Console.WriteLine("Enter column to remove.");
+            oldColumn = int.Parse(Console.ReadLine());
+            validationChecker = SelectChecker(oldRow, oldColumn);
+
+            if(Checkers.Contains(validationChecker))
+            {
+                RemoveChecker(oldRow,oldColumn);
+                reloadBoard();
+            }
+            
+
+
+        }
+
+        //finds checker passed in and removes it from the list of board checkers
+        
+        public void RemoveChecker(int row, int column)
+        {
+            Checker removedChecker ;
+            removedChecker = SelectChecker(row, column);
+            Checkers.Remove(removedChecker); //removes the checker object that was returned at that location provided
+        }
+        
+        public bool CheckForWin()
+        {
+            return Checkers.All(x => x.Color == "white") || !Checkers.Exists(x => x.Color == "white");
```

Comment:
Slight performance improvement here. `.All(...)` will always require looping over every item, where as a negated `.Exists(...)` will stop when it finds the first matching item.